File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.java
import java.util.*;

class Solution {
    
    static int parent[];
    
    public int solution(int n, int[][] costs) {
        
        Arrays.sort(costs, new Comparator<int[]>(){
            @Override
            public int compare(int[] o1, int[] o2){
                return o1[2] - o2[2];
            }
        });
        
        parent = new int[n+1];
        
        for(int i=1; i<n+1; i++){
            parent[i] = i;
        }
        
        int answer = 0;
        
        for(int i=0; i<costs.length; i++){
            int from = costs[i][0];
            int to = costs[i][1];
            if(find(from) == find(to)) continue;
            answer += costs[i][2];
            union(from, to);
        }
        
        return answer;
    }
    
    public int find(int node){
        if(parent[node] == node) return node;
        return parent[node] = find(parent[node]);
    }
    
    public void union(int a, int b){
        int pa = parent[a];
        int pb = parent[b];
        if(pa != pb) parent[pb] = pa;
    }

}

File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.js

const parent = [];

function solution(n, costs) {
    
    costs.sort((a, b) => a[2]-b[2]);
    
    for(let i=0; i<n+1; i++){
        parent.push(i);
    }
    
    var answer = 0;
    
    for(let i=0; i<costs.length; i++){
        var from = costs[i][0];
        var to = costs[i][1];
        if(find(from) == find(to)) continue;
        answer += costs[i][2];
        union(from, to);
    }
    
    return answer;
}

function find(node){
    if(parent[node] == node) return node;
    return parent[node] = find(parent[node]);
}

function union(a , b){
    let pa = parent[a];
    let pb = parent[b];
    if(pa != pb) parent[pb] = pa;
}
